### DISCUSSION TITLE
Cookies toolbar.

### DISCUSSION THREAD

**Discussion by Avskum** [mediacms-io/mediacms#797] [Feature Requests]
In EU we have mandatory cookies toolbar on website which should inform user about that you are using cookies. Is there possibile to give us oportunity to enable this toolbar in admin panel?

**Comment by mgogoulos** [comment]
Hi, this makes sense, and would be good to have a baseline configuration with messages and options. 
On a side note, https://demo.mediacms.io/ is not having cookies notification since it does not call external scripts and APIs (there's no analytics, trackers, ads etc)

**Comment by Avskum** [comment]
Sadly even if you are not using in reality any cookies, law enforcements can be very strict sometime. Not sure how GDPR law is handling situations like this, its always better to have this alert on your website to be honest. 

I want to fight against desinformations in my country via mediacms and sadly I cant start website, because of this missing feature, because I am expecting regular notyfing about my website if not using cookies to our local law enforcers.

**Reply by mgogoulos** [reply]
I see. There are lots of services that can help you get the code (html, css and some javascript) for a cookie consent banner. Once you have the code, edit file `templates/root.html` , place the code inside the <body> section, and finally restart mediacms. Hope this helps

**Reply by Avskum** [reply]
Sadly I dont know how to do it as I am very lame with django/python.I will wait for official release and will check other possibilities..

**Reply by mgogoulos** [reply]
I've just added a simple Cookie on https://demo.mediacms.io/ , check it out and if this works for you here's how to add it:

1. Open file templates/base.html and add the following text below
```
{% include "components/header.html" %}
```

```
<script src="https://cdn.websitepolicies.io/lib/cookieconsent/cookieconsent.min.js" defer></script><script>window.addEventListener("load",function(){window.wpcc.init({"border":"normal","corners":"normal","colors":{"popup":{"background":"#222222","text":"#ffffff","border":"#FF8000"},"button":{"background":"#FF8000","text":"#000000"}},"position":"top-right","content":{"message":"Hi there, we are using cookies on this website. We don't use tracking or analytics here, just the essentials for the Website to work.\n","button":"Understood! Yum!","link":"Click here to learn more."}})});</script>
```

2. Save the file, and restart mediacms `systemctl restart mediacms`

This is very minimal but depending on what you are offering it might work. Using a service like the above creates the javascript code that is needed (based on your colours/options) and you just have to paste it on the above section, no need for python/django work here!

**Comment by mgogoulos** [comment]
I've added a sample cookie consent (that is off by default but is trivial to enable), check out https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md#17-cookie-consent-code

**Reply by Avskum** [reply]
Thanks! This works perfect, tested and no issues there :-)
